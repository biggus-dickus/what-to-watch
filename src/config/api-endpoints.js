export const BASE_HOST = `https://es31-server.appspot.com`;

export const ApiEndpoint = {
  BASE_URL: `${BASE_HOST}/wtw`,
  COMMENTS: `/comments`,
  FILMS: `/films`,
  FAVOURITE: `favorite`,
  LOGIN: `/login`
};
