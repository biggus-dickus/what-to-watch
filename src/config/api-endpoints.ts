export const BASE_HOST: string = `https://es31-server.appspot.com`;

export enum ApiEndpoint {
  BASE_URL = 'https://es31-server.appspot.com/wtw',
  COMMENTS = '/comments',
  FILMS = '/films',
  FAVOURITE = 'favorite',
  LOGIN = '/login',
  PROMO = '/films/promo'
}
