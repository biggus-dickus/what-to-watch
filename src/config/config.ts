// How much time to wait in ms before playing a film preview
export const HOVER_DELAY = 1000;

// How many films to render in "More like this" section (film page)
export const MORE_LIKE_THIS_LIMIT = 4;

// How many films to display at once on mount
export const FILMS_LIMIT = 20;

// How many genres to display at once
export const GENRES_LIMIT = 9 + 1; // 9 + "All genres"

// Navigation tabs on film page
export const NAV_OVERVIEW_ID = `overview`;
export const NAV_DETAILS_ID = `details`;
export const NAV_REVIEWS_ID = `reviews`;
export const filmTabs = [
  {
    id: NAV_OVERVIEW_ID,
    text: `Overview`
  },
  {
    id: NAV_DETAILS_ID,
    text: `Details`
  },
  {
    id: NAV_REVIEWS_ID,
    text: `Reviews`
  }
];

// Film review constraints
export const REVIEW_CHARS_MIN = 50;
export const REVIEW_CHARS_MAX = 400;
