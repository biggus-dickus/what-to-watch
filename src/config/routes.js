const Route = Object.freeze({
  ADD_REVIEW: `/film/:id/review`,
  FILM: `/film/:id`,
  INDEX: `/`,
  MY_LIST: `/mylist`,
  SIGN_IN: `/login`
});

export default Route;

export const isPrivateRoute = (pathname) => {
  const myListRe = /\/mylist/;
  const addReviewRe = /\/film\/[\d]\/review/;

  return myListRe.test(pathname) || addReviewRe.test(pathname);
};
