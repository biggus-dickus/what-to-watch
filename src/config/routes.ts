enum Route {
  ADD_REVIEW = '/film/:id/review',
  FILM = '/film/:id',
  INDEX = '/',
  MY_LIST = '/mylist',
  SIGN_IN = '/login'
}

export default Route;

export const isPrivateRoute = (pathname: string): boolean => {
  const myListRe = /\/mylist/;
  const addReviewRe = /\/film\/[1-9][\d]*\/review/;

  return myListRe.test(pathname) || addReviewRe.test(pathname);
};
