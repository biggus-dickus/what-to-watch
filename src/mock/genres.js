export const genres = [`Comedies`, `Crime`, `Documentary`, `Dramas`, `Horror`, `Kids & Family`, `Romance`, `Sci-Fi`, `Thrillers`, `Good ol' Porn`];
