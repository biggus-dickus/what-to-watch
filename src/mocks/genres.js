export const Genre = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};
