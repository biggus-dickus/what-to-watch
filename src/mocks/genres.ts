import {Genre as GenreType} from '../types'; // eslint-disable-line

export const Genre = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};

export const mockGenres = Object.values(Genre);

export const mockGenreEntity: GenreType = {
  currentGenre: Genre.CRIME,
  genres: [...mockGenres],
  onGenreChange: jest.fn()
};
