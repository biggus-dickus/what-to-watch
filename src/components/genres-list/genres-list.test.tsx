import * as React from 'react';
import * as renderer from 'react-test-renderer';
import * as Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';

import {GENRES_LIMIT} from '../../config/config';
import {mockGenres} from '../../mocks/genres';

import GenresList from './genres-list';

configure({adapter: new Adapter()});


const props = {
  genres: mockGenres,
  currentGenre: mockGenres[1],
  onGenreChange: jest.fn()
};

describe(`GenresList test suite`, () => {
  it(`<GenresList /> should render correctly`, () => {
    const tree = renderer.create(<GenresList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`should display no more genres than set by config`, () => {
    const wrapper = shallow(<GenresList {...props} />);
    const genreElems = wrapper.find(`[data-test="at-genres-list"]`).children();

    expect(genreElems.length).toBeLessThanOrEqual(GENRES_LIMIT);
  });
});
