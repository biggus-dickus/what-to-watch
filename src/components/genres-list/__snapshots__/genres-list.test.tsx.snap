// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenresList /> should render correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#All-genres"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href={null}
      onClick={[Function]}
    >
      Comedies
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Crime"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Documentary"
      onClick={[Function]}
    >
      Documentary
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Dramas"
      onClick={[Function]}
    >
      Dramas
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Horror"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Kids-&-Family"
      onClick={[Function]}
    >
      Kids & Family
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Romance"
      onClick={[Function]}
    >
      Romance
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Sci-Fi"
      onClick={[Function]}
    >
      Sci-Fi
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Thrillers"
      onClick={[Function]}
    >
      Thrillers
    </a>
  </li>
</ul>
`;

exports[`GenresList test suite <GenresList /> should render correctly 1`] = `
<ul
  className="catalog__genres-list"
  data-test="at-genres-list"
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#All-genres"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href={null}
      onClick={[Function]}
    >
      Comedies
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Crime"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Documentary"
      onClick={[Function]}
    >
      Documentary
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Dramas"
      onClick={[Function]}
    >
      Dramas
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Horror"
      onClick={[Function]}
    >
      Horror
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Kids-&-Family"
      onClick={[Function]}
    >
      Kids & Family
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Romance"
      onClick={[Function]}
    >
      Romance
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Sci-Fi"
      onClick={[Function]}
    >
      Sci-Fi
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#Thrillers"
      onClick={[Function]}
    >
      Thrillers
    </a>
  </li>
</ul>
`;
